snuba/request_schema.py

```diff
@@ -9,13 +9,14 @@
 
 from snuba.query.extensions import QueryExtension
 from snuba.query.query import Query
-from snuba.query.schema import GENERIC_QUERY_SCHEMA
+from snuba.query.schema import GENERIC_QUERY_SCHEMA, QUERY_SETTINGS_SCHEMA
 from snuba.schemas import Schema, validate_jsonschema
 
 
 @dataclass(frozen=True)
 class Request:
     query: Query
+    query_settings: Mapping[str, str]
```

Comment:
I think these are request settings, though? Since they influence things like the extensions (and likely other things like the query execution strategy with the splitter) which are not part of the "big-Q" `Query` itself.